docs/data/data-grid/filtering-recipes/FilterPresetsPanel.tsx

```diff
@@ -0,0 +1,369 @@
+    presetsStore.subscribe,
+    presetsStore.getSnapshot,
+    presetsStore.getServerSnapshot,
+  );
+
+  const store = React.useMemo(() => {
+    try {
+      return JSON.parse(storeString) as FilterPresetStore;
+    } catch {
+      return EMPTY_STORE;
+    }
+  }, [storeString]);
+
+  return React.useMemo(
+    () => [store, presetsStore.update] as const,
+    [store, presetsStore.update],
+  );
+};
+
+type GridFilterPanelProps = React.ComponentProps<typeof GridFilterPanel>;
+
+function CustomFilterPanel(props: GridFilterPanelProps) {
+  const apiRef = useGridApiContext();
+  const [store, setStore] = usePersistedPresets();
+  const [createFilterDialogOpen, setCreateFilterDialogOpen] = React.useState(false);
+  const [createFilterName, setCreateFilterName] = React.useState('');
+
+  const getCurrentFilterModel = () => apiRef.current.state.filter.filterModel;
+  const hasActiveFilters = () => getCurrentFilterModel().items.length > 0;
+
+  const handleSavePreset = () => {
+    if (!store.activePresetId) {
+      return;
+    }
+
+    const currentFilterModel = getCurrentFilterModel();
+
+    setStore({
+      ...store,
+      presets: store.presets.map((p) =>
+        p.id === store.activePresetId
+          ? {
+              ...p,
+              filterModel: currentFilterModel,
+            }
+          : p,
+      ),
+    });
+  };
+
+  const handleCreateFilter = () => {
+    if (!createFilterName.trim()) {
+      return;
+    }
+
+    const newPreset: FilterPreset = {
+      id: `preset_${Date.now()}`,
+      name: createFilterName.trim(),
+      filterModel: getCurrentFilterModel(),
+      createdAt: new Date().toISOString(),
+    };
+
+    setStore({
+      ...store,
+      presets: [...store.presets, newPreset],
+      activePresetId: newPreset.id,
+    });
+
+    setCreateFilterDialogOpen(false);
+    setCreateFilterName('');
+  };
+
+  const handleLoadPreset = (presetId: string) => {
+    const preset = store.presets.find((p) => p.id === presetId);
+    if (preset) {
+      apiRef.current.setFilterModel(preset.filterModel);
+
+      setStore({
+        ...store,
+        activePresetId: presetId,
+      });
+    }
+  };
+
+  const handleDeletePreset = (presetId: string) => {
+    setStore({
+      ...store,
+      presets: store.presets.filter((p) => p.id !== presetId),
+      activePresetId:
+        store.activePresetId === presetId ? null : store.activePresetId,
+    });
+  };
+
+  return (
+    <React.Fragment>
+      <Box
+        sx={{
+          minWidth: 256,
+          display: 'flex',
+          flexDirection: 'column',
+          'div:has(&)': { flexWrap: 'wrap' },
+        }}
+      >
+        <Box
+          sx={{
+            p: 2,
+            pb: 1,
+            display: 'flex',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+          }}
+        >
+          <Typography variant="subtitle2" sx={{ fontWeight: 'bold' }}>
+            Saved Filters
```

Comment:
This is clearer to me, wdyt?
```suggestion
            Saved Filter Presets
```